src/Compilers/CSharp/Portable/Symbols/Retargeting/RetargetingSymbolTranslator.cs

```diff
@@ -184,6 +184,15 @@ private NamedTypeSymbol RetargetNamedTypeDefinition(NamedTypeSymbol type, Retarg
             {
                 Debug.Assert(type.IsDefinition);
 
+                if (type.IsNativeIntegerType)
+                {
+                    if (!SymbolMap.TryGetValue(type, out var retargeted))
+                    {
+                        retargeted = SymbolMap.GetOrAdd(type, Retarget(type.NativeIntegerUnderlyingType, options).AsNativeInteger());
```

Comment:
>options [](start = 105, length = 7)

Should we pass RetargetOptions.RetargetPrimitiveTypesByTypeCode here to force lookup as special type regardless of the ```options```?

Reply:
Otherwise things might fall apart if name of the core library is changed.

---
In reply to: [393271200](https://github.com/dotnet/roslyn/pull/42419#discussion_r393271200) [](ancestors = 393271200)

Reply:
It looks like we can hit this scenario when native int type is used as base type (an error case, but still something that should be handled reasonably well). It feels like we don't want to force-change the options in this case (to be consistent with what we do for [U]IntPtr references), but it is not clear whether it is appropriate to create a native int wrapper symbol. Will the new underlying symbol even be able able to create one without asserting or failing otherwise? Let's try to come up with a test case.

---
In reply to: [393276776](https://github.com/dotnet/roslyn/pull/42419#discussion_r393276776) [](ancestors = 393276776,393271200)

Reply:
Added `Retargeting_05` although it doesn't hit this code path.

---
In reply to: [393281988](https://github.com/dotnet/roslyn/pull/42419#discussion_r393281988) [](ancestors = 393281988,393276776,393271200)

Reply:
> Added Retargeting_05 although it doesn't hit this code path.

Well, we need to find a way to hit it. One of the previous comments on this thread might help: "Otherwise things might fall apart if name of the core library is changed."  Also, it is not clear what "it doesn't hit this code path" means exactly. Are we not doing retargeting at all? This feels suspicious.

---
In reply to: [394423057](https://github.com/dotnet/roslyn/pull/42419#discussion_r394423057) [](ancestors = 394423057,393281988,393276776,393271200)